### DISCUSSION TITLE
Improve combinatorics functionality in mathnet-numerics

### DISCUSSION THREAD

**Discussion by s952163** [mathnet/mathnet-numerics#797] [Ideas]
This is a feature request to improve combinatorics related functionality in MathNet.Numerics. The [combinatorics namespace](https://numerics.mathdotnet.com/api/MathNet.Numerics/Combinatorics.htm) does contain the functions to calculate the number of possible combinations/variations/permutations. It also contains the functions to generate a random sample of those. However unfortnately it lacks functionality to actually generate all the combinations etc. For example see this [SO post](http://stackoverflow.com/questions/42184242/f-9-names-into-disjointed-groups-of-2-3-and-4), so one ends up using recursion or list comprehensions. F# 4.1 has added `List.allPairs` for the Cartesian product.

As this is the go-to library for statistical and numerical analysis it would be great if Mathnet.Numerics could generate the actual variations/permutations/combinations with and without repetitions. In .NET currently this library does: https://github.com/eoincampbell/combinatorics . For examples see: https://www.codeproject.com/Articles/26050/Permutations-Combinations-and-Variations-using-C-G and [Combinatorics in .NET Part 1](http://trycatch.me/combinatorics-in-net-part-i-permutations-combinations-variations/).

**Comment by ademchenko** [comment]
Is there any movement of that discussion?
